PowerShellGet/PSModule.psm1

```diff
@@ -13803,6 +13887,61 @@ function Get-PrivateData
     }
 }
 
+function Validate-PreReleaseString 
+{
+    [CmdletBinding(PositionalBinding=$false)]
+    Param 
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Version,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $PreRelease,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [Object]
+        $InfoObject
+    )
+
+    # Remove leading hyphen (if present)
+    if ( $PreRelease -and $PreRelease.StartsWith('-') )
+    {
+        $PreRelease = $PreRelease.Substring(1)
+    }
+
+    # only these characters are allowed in a prerelease string
+    $validCharacters = "^[a-zA-Z0-9]+$"
+    $prereleaseStringValid = $PreRelease -match $validCharacters
+    if (-not $prereleaseStringValid)
+    {
+        ThrowError -ExceptionName "System.InvalidOperationException" `
+                   -ExceptionMessage $LocalizedData.InvalidCharactersInPreReleaseString `
+                   -ErrorId "InvalidCharactersInPreReleaseString" `
+                   -CallerPSCmdlet $PSCmdlet `
+                   -ErrorCategory InvalidOperation `
+                   -ExceptionObject $InfoObject
+    }
+
+    # Validate that Version contains at least 3 parts
+    $versionPartsCount = $Version.Split('.').Count
+    if ( $versionPartsCount -lt 3 )

```

Comment:
Good point, it looks like it is minimum 3 parts but maximum 4 parts.  I will add another validation check.